agent/reconciler/host_reconciler.go

```diff
@@ -116,13 +135,7 @@ func (r *HostReconciler) reconcileNormal(ctx context.Context, byoHost *infrastru
 
 		if r.SkipK8sInstallation {
 			logger.Info("Skipping installation of k8s components")
-		} else if r.UseInstallerController {
-			if byoHost.Spec.InstallationSecret == nil {
-				logger.Info("K8sInstallationSecret not ready")
-				conditions.MarkFalse(byoHost, infrastructurev1beta1.K8sNodeBootstrapSucceeded, infrastructurev1beta1.K8sInstallationSecretUnavailableReason, clusterv1.ConditionSeverityInfo, "")
-				return ctrl.Result{}, nil
-			}
-		} else {
+		} else if !r.UseInstallerController {
```

Comment:
Instead of having a separate if block for `if r.UseInstallerController && !conditions.IsTrue(byoHost, infrastructurev1beta1.K8sComponentsInstallationSucceeded)`, why don't we use the existing if...else if...else blocks?

Reply:
@anusha94 I have moved inside if..else block. Thanks